### ISSUE TITLE
duplicate attribute name

### ISSUE THREAD

**Issue by s9105947** [ComputationalRadiationPhysics/picongpu#4072]
Dear Maintainers,

during [the work on PICMI](https://github.com/s9105947/picongpu/tree/picmi-integration) the code generation was off and triggered a uncaught configuration error.

## Description
Define an particle species attribute **twice** (with the same name) for the same species:

```c++
// full code (wrongly) generated

                value_identifier(float_X, MassRatio_species_H, 1.6738233791328002e-27 / SI::BASE_MASS_SI);
                value_identifier(float_X, ChargeRatio_species_H, 1.6021766339999999e-19 / SI::BASE_CHARGE_SI);

        using ParticleFlags_species_H = MakeSeq_t<
                    massRatio<MassRatio_species_H>,
                    chargeRatio<ChargeRatio_species_H>,


                    atomicNumbers<ionization::atomicNumbers::Hydrogen_t>,
                    ionizationEnergies<ionization::energies::AU::Hydrogen_t>,

            particlePusher<UsedParticlePusher>,
            shape<UsedParticleShape>,
            interpolation<UsedField2Particle>,
            current<UsedParticleCurrentSolver>>;

        using ParticleAttributes_species_H = MakeSeq_t<
            momentum,
                position<position_pic>
                    ,
                weighting
                    ,
                momentum
            >;

        using species_H = Particles<
            PMACC_CSTRING("H"),
            ParticleFlags_species_H,
            ParticleAttributes_species_H>;



    using VectorAllSpecies = MakeSeq_t<
            species_H
        >;
```

Note that `momentum` is listed twice.

> Macroparticles of the species **must be created** in the init pipeline, otherwise the behavior will not occur.

## Expected Behavior
compilation error "duplicate names are forbidden"

## Actual Behavior
compilation works fine, but execution fails **after** initialization with:

```
initialization time: 10sec 393msec = 10.393 sec
double free or corruption (!prev)
```

According to debugging output, this occurs during writing in OpenPMD:

```
[...]
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PMaccVerbose UNDEFINED_LVL(80) | create event
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD: write field: H_all_chargeDensity 1 0x2b85000
PMaccVerbose MEMORY(1) | DataConnector: sharing access to 'FieldTmp0' (1 uses)
PMaccVerbose MEMORY(1) | DataConnector: sharing access to 'H' (1 uses)
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD: write field: H_all_energyDensity 1 0x2b85000
PMaccVerbose MEMORY(1) | DataConnector: sharing access to 'FieldTmp0' (1 uses)
PMaccVerbose MEMORY(1) | DataConnector: sharing access to 'H' (1 uses)
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD: write field: H_all_particleMomentumComponent 1 0x2b85000
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD: ( end ) writing fields.
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD: (begin) writing particle species.
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD: (begin) write species: H
PMaccVerbose MEMORY(1) | DataConnector: sharing access to 'H' (1 uses)
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:   (begin) count particles: H
PMaccVerbose MEMORY(1) | Create device 1D data: 0 MiB
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:   ( end ) count particles: H = 13824
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:   (begin) malloc host memory: H
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:   ( end ) malloc host memory: H
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:   (begin) copy particle host (with hierarchy) to host (without hierarchy): H
PMaccVerbose MEMORY(1) | DataConnector: sharing access to 'MallocMCBuffer' (1 uses)
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  (begin) write particle records for H
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  (begin) write species attribute: momentum
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  ( end ) write species attribute: momentum
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  (begin) write species attribute: position
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  ( end ) write species attribute: position
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  (begin) write species attribute: weighting
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  ( end ) write species attribute: weighting
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  (begin) write species attribute: momentum
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  ( end ) write species attribute: momentum
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  (begin) write species attribute: totalCellIdx
double free or corruption (!prev)
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  ( end ) write species attribute: totalCellIdx
PIConGPUVerbose INPUT_OUTPUT(32) | openPMD:  (begin) free memory: H
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpiexec noticed that process rank 0 with PID 0 on node fwk394 exited on signal 6 (Aborted).
--------------------------------------------------------------------------
```

**Comment by psychocoderHPC** [comment]
Thanks for reporting. This is a very interesting bug. 
We need to check https://github.com/ComputationalRadiationPhysics/picongpu/blob/beba68a3a2f9736ec487195e066a3ae598a4e50b/include/pmacc/particles/memory/buffers/ParticlesBuffer.hpp#L86-L90 that all attributes in the list are unique.

**Comment by sbastrakov** [comment]
Solved with #4078.
